cmd/kube-proxy/app/server_windows.go

```diff
@@ -81,7 +82,8 @@ func (s *ProxyServer) platformCheckSupported() (ipv4Supported, ipv6Supported, du
 // createProxier creates the proxy.Provider
 func (s *ProxyServer) createProxier(config *proxyconfigapi.KubeProxyConfiguration, dualStackMode, initOnly bool) (proxy.Provider, error) {
 	if initOnly {
-		return nil, fmt.Errorf("--init-only is not implemented on Windows")
+		klog.Info("--init-only is not implemented on Windows")
+		return nil, nil
```

Comment:
what happens in the code a path if a nil proxier without an error is returned?
shouldn't this be handled with an additional check somewhere so that there are no panics?

Reply:
I got a contradicting comment from @thockin https://github.com/kubernetes/kubernetes/pull/120864#discussion_r1369481811. @neolit123 please coordinate with him and I do whatever you think is best

Reply:
I don't mind the exact implementation details. IMO, what makes sense for Windows and the multi-os portability:
- the flag is passed, kube-proxy shows a message and exits with status 0. status 1 is more accurate, but this is done to not trip the init container setup.
-  the flag is not passed, kube-proxy runs in priv mode as it does today.

Reply:
mentioned this ^ in #sig-windows on the k8s slack:
https://kubernetes.slack.com/archives/C0SJ4AFB7/p1698317022006619